wandb/sdk/artifacts/artifact.py

```diff
@@ -1647,12 +1647,17 @@ def download(
 
         Raises:
             ArtifactNotLoggedError: If the artifact is not logged.
+            RuntimeError: If the artifact is attempted to be downloaded in offline mode.
         """
         self._ensure_logged("download")
 
         root = FilePathStr(str(root or self._default_root()))
         self._add_download_root(root)
 
+        # TODO: we need a better way to check for offline mode across the app, as this is an anti-pattern
+        if env.is_offline() or util._is_offline():
+            raise RuntimeError("Cannot download artifacts in offline mode.")
```

Comment:
is it a runtimeError or a usageError?

Reply:
From the other usages across the codebase this fits the RuntimeError the most.